### DISCUSSION TITLE
Messagable::threads() does not respect soft deletes - so how do I truly remove a participant?

### DISCUSSION THREAD

**Discussion by cyrillkalita** [cmgmyr/laravel-messenger#376] [Q&A] [answered]
Could be a silly question, really, but `BelongsToMany` does not account for `SoftDeletes`.

So a test structured like this, fails:
- create a thread,
- add a user participant to a thread
- remove user participant using `removeParticipant()`

The pivot attribute deleted_at is populated, as expected.
so:
1. thread->participants()-count() is 0, as expected, but
2. $user->threads()->count() still returns 1, because this is how `BelongsToMany` works

I read #205 and cannot help but wonder: am I missing something obvious?

**Comment by cmgmyr** [comment, marked as answer]
@cyrillkalita thanks for reporting. You are right, those relationships don't respect soft deletes. I'm surprised that 1. this was missed, and 2. no one has reported this sooner haha.

I moved this to issue #379 and a PR fix in #380. Would you give it a quick glance to double I covered everything? 🙏

**Reply by cmgmyr** [reply]
tagged `2.26.0` with the fix
